chord_metadata_service/mcode/parse_fhir_mcode.py

```diff
@@ -0,0 +1,306 @@
+        return ontology_value
+    # will be raised if there is no "code" in Coding element
+    except KeyError as e:
+        raise KeyError(e)
+
+
+def patient_to_individual(resource):
+    """ Patient to Individual. """
+
+    # TODO when smart-on-fhir for fhir 4.0 is issued change it for  function form fhir-utils
+    individual = {
+        "id": resource["id"]
+    }
+    if "identifier" in resource:
+        individual["alternate_ids"] = [alternate_id["value"] for alternate_id in resource["identifier"]]
+    gender_to_sex = {
+        "male": "MALE",
+        "female": "FEMALE",
+        "other": "OTHER_SEX",
+        "unknown": "UNKNOWN_SEX"
+    }
+    if "gender" in resource:
+        individual["sex"] = gender_to_sex[resource["gender"]]
+    if "birthDate" in resource:
+        individual["date_of_birth"] = resource["birthDate"]
+    if "active" in resource:
+        if resource["active"]:
+            individual["active"] = True
+    if "deceasedBoolean" in resource:
+        individual["deceased"] = True
+    return individual
+
+
+def observation_to_labs_vital(resource):
+    """ Observation with tumor marker to LabsVital. """
+    labs_vital = {
+        "id": resource["id"]
+    }
+    if "code" in resource:
+        labs_vital["tumor_marker_code"] = get_ontology_value(resource, "code")
+    if "valueCodeableConcept" in resource:
+        labs_vital["tumor_marker_data_value"] = get_ontology_value(resource, "valueCodeableConcept")
+    if "subject":
+        labs_vital["individual"] = resource["subject"]["reference"].split("uuid:")[-1]
+    return labs_vital
+
+
+def observation_to_tnm_staging(resource):
+    """ Observation with tnm staging to TNMStaging. """
+    tnm_staging = {
+        "id": resource["id"],
+        "tnm_staging_value": {}
+    }
+    if "valueCodeableConcept" in resource:
+        tnm_staging["tnm_staging_value"]["data_value"] = get_ontology_value(resource, "valueCodeableConcept")
+    if "method" in resource:
+        tnm_staging["tnm_staging_value"]["staging_system"] = get_ontology_value(resource, "method")
+    # reference to Condition
+    if "focus" in resource:
+        tnm_staging["cancer_condition"] = resource["focus"][0]["reference"].split("/")[-1]
+    return tnm_staging
+
+
+def procedure_to_crprocedure(resource):
+    """ Procedure to Cancer related Procedure. """
+
+    cancer_related_procedure = {
+        "id": resource["id"]
+    }
+    if "code" in resource:
+        cancer_related_procedure["code"] = get_ontology_value(resource, "code")
+    if "bodySite" in resource:
+        cancer_related_procedure["body_site"] = get_ontology_value(resource, "bodySite")
+    if "reasonCode" in resource:
+        codes = []
+        for code in resource["reasonCode"]["coding"]:
+            reason_code = {
+                "id": f"{code['system']}:{code['code']}",
+                "label": f"{code['display']}"
+            }
+            codes.append(reason_code)
```

Comment:
this might be better written as a list comprehension, but you don't need to change it